cast/src/starknet_commands/declare.rs

```diff
@@ -15,41 +15,112 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::{Command, Stdio};
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetArtifacts {
+    version: u32,
+    contracts: Vec<ScarbStarknetContract>,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetContract {
+    id: String,
+    package_name: String,
+    contract_name: String,
+    artifacts: ScarbStarknetContractArtifact,
+}
+
+#[allow(dead_code)]
+#[derive(Deserialize)]
+struct ScarbStarknetContractArtifact {
+    sierra: PathBuf,
+    casm: Option<PathBuf>,
```

Comment:
Generally speaking, `sierra` is also optional.

Reply:
I think we will never have a case where we want `sierra=false` in starknet target. I added a description for better error handling in case a user makes changes to Scarb.toml in https://github.com/software-mansion/protostar/issues/2141

Reply:
This change is taken from https://github.com/software-mansion/protostar/pull/2120/files#diff-e94fd807a4275d652e6b7372b7c207fd996d38ae3250db9eb18b1b63cc511472R104-R107

1. You should've noticed that in https://github.com/software-mansion/protostar/pull/2120 (I see you didn't do a review but that's not my fault after all).
2. We should have a consistent code base. Therefore if I do this change here, it will not be consistent. If I decide to go through the code and do this change also in `protostar-rust/src/cheatcodes_hint_processor.rs`, I will have to consider if that is a good change, debug stuff, etc. So I do not think it should be done here.

I think we should leave it as is. If you feel this is very bad, feel free to create an issue for this, thanks.